pkg/cfn/builder/api_test.go

```diff
@@ -1136,6 +1137,49 @@ var _ = Describe("CloudFormation template builder API", func() {
 
 	})
 
+	Context("NodeGroupAppMeshPreview", func() {
+		cfg, ng := newClusterConfigAndNodegroup(true)
+
+		ng.IAM.WithAddonPolicies.AppMeshPreview = api.Enabled()
+
+		build(cfg, "eksctl-test-appmesh-preview", ng)
+
+		roundtrip()
+
+		It("should have correct policies", func() {
+			Expect(ngTemplate.Resources).To(HaveKey("PolicyAppMesh"))
+
+			policy3 := ngTemplate.Resources["PolicyAppMesh"].Properties
+
+			Expect(policy3.Roles).To(HaveLen(1))
+			isRefTo(policy3.Roles[0], "NodeInstanceRole")
+
+			Expect(policy3.PolicyDocument.Statement).To(HaveLen(1))
+			Expect(policy3.PolicyDocument.Statement[0].Effect).To(Equal("Allow"))
+			Expect(policy3.PolicyDocument.Statement[0].Resource).To(Equal("*"))
+			Expect(policy3.PolicyDocument.Statement[0].Action).To(Equal([]string{
+				"servicediscovery:CreateService",
+				"servicediscovery:DeleteService",
+				"servicediscovery:GetService",
+				"servicediscovery:GetInstance",
+				"servicediscovery:RegisterInstance",
+				"servicediscovery:DeregisterInstance",
+				"servicediscovery:ListInstances",
+				"servicediscovery:ListNamespaces",
+				"servicediscovery:ListServices",
+				"servicediscovery:GetInstancesHealthStatus",
+				"servicediscovery:UpdateInstanceCustomHealthStatus",
+				"servicediscovery:GetOperation",
+				"route53:GetHealthCheck",
+				"route53:CreateHealthCheck",
+				"route53:UpdateHealthCheck",
+				"route53:ChangeResourceRecordSets",
+				"route53:DeleteHealthCheck",
```

Comment:
Can we extract these into a common array for both tests so it's easier to see in a quick look what is the difference between appmesh and appmesh-preview?

Reply:
Done!